akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1983,6 +1984,30 @@ trait FlowOps[+Out, +Mat] {
   def keepAlive[U >: Out](maxIdle: FiniteDuration, injectedElem: () ⇒ U): Repr[U] =
     via(new Timers.IdleInject[Out, U](maxIdle, injectedElem))
 
+  /**
+   * Sends elements from buffer if upstream does not emit for a configured amount of time. In other words, this
+   * stage attempts to maintains a base rate of emitted elements towards the downstream using elements from upstream.
+   *
+   * If upstream emits new elements until the accumulated elements in the buffer exceed the specified minimum size
+   * used as the keep alive elements, then the base rate is no longer maintained until we reach another period without
+   * elements form upstream.
+   *
+   * The keep alive period is the size times the interval.
+   *
+   * '''Emits when''' upstream emits an element or if the upstream was idle for the configured period
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @see [[keepAlive]]
+   * @see [[expand]]
+   */
+  def keepAliveFromBuffer[U >: Out](size: Int, interval: FiniteDuration, extrapolate: U ⇒ List[U]): Repr[U] =
```

Comment:
from buffer naming is somewhat misleading... fit it into existing terminology where where such flattening is referred to as concat? `keepAliveConcat`?

Reply:
has to be replicated to javadsl as well -- `akka/stream/javadsl/Flow.scala`

Reply:
`extrapolate: U ⇒ List[U]` means it can't do it's keep-alive job if upstream never emited an element, should we consider another function signature here?

Reply:
`immutable.Seq` is preferred rather than `List` when accepting collections